README.md

```diff
@@ -3,31 +3,82 @@
 Provides utilities for collecting metrics from a Tokio application, including
 runtime and per-task metrics.
 
-## Getting Started
+```toml
+[dependencies]
+tokio-metrics = { version = "0.1.0", default-features = false }
+```
 
-From within a Tokio application, use `TaskMetrics` to instrument tasks before
-spawning them and to observe the metrics. All tasks instrumented with a given
-`TaskMetrics` instance will aggregate their metrics together. To split out
-metrics for different tasks, use separate `TaskMetrics` instances.
+## Getting Started With Task Metrics
 
-```rust
-// Create a `TaskMetrics` instance
-let metrics = tokio_metrics::TaskMetrics::new();
+Use `TaskMonitor` to instrument tasks before spawning them, and to observe
+metrics for those tasks. All tasks instrumented with a given `TaskMonitor`
+aggregate their metrics together. To split out metrics for different tasks, use
+separate `TaskMetrics` instances.
 
-// Instrument any number of tasks using the `TaskMetrics` instance
-tokio::spawn(metrics.instrument(async {
-    // Do work here
-}));
+```rust
+// construct a TaskMonitor
+let monitor = tokio_metrics::TaskMonitor::new();
 
-// Use the `TaskMetrics` instance to observe reported data.
-for sample in metrics.sample() {
-    println!("Metrics = {:?}", sample);
+// print task metrics every 500ms
+{
+    let frequency = std::time::Duration::from_millis(500);
+    let monitor = monitor.clone();
+    tokio::spawn(async move {
+        for metrics in monitor.intervals() {
+            println!("{:?}", metrics);
+            tokio::time::sleep(frequency).await;
+        }
+    });
+}
 
-    // Wait some time before sampling metrics again.
-    tokio::time::sleep(Duration::from_millis(100)).await;
+// instrument some tasks and await them
+loop {
+    monitor.instrument(do_work()).await;
```

Comment:
I would always show this inside of a `tokio::spawn(monitor.instrument(....))` to hammer that pattern.